### PR TITLE
app-vim/rust-vim: bump to 20210215

### PR DESCRIPTION
**Pull request by scardracs**
Package-Manager: Portage-3.0.20, Repoman-3.0.3
Acked-by: Patrice Clement <monsieurp@gentoo.org>
Signed-off-by: Marco Scardovi <marco@scardovi.com>

### CODE REVIEW

**Comment by scardracs** [comment]
@monsieurp if you are ok with it I'll ack you here 😄

File: app-vim/rust-vim/rust-vim-1_pre20210215.ebuild [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+inherit vim-plugin
+
+COMMIT="87c745d8d506fc1eecc1d81df15d5bde1658a2fc"
+MY_PN="${PN/-/.}"
+
+DESCRIPTION="Vim configuration for Rust"
+HOMEPAGE="https://www.rust-lang.org https://github.com/rust-lang/rust.vim"
+SRC_URI="https://github.com/rust-lang/rust.vim/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
```
Comment (ionenwks) [inline app-vim/rust-vim/rust-vim-1_pre20210215.ebuild:15]: The previous `|| ( MIT Apache-2.0 )` seems right to me, same that dev-lang/rust uses -- you'll see this package has both license files.

I know github lists Apache-2.0, but github doesn't really care for nuances.

File: app-vim/rust-vim/metadata.xml [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+	</maintainer>
+	<maintainer type="project">
+		<email>vim@gentoo.org</email>
+		<name>Gentoo Vim Project</name>
+	</maintainer>
+	<maintainer type="project" proxied="proxy">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
 	<upstream>
```
Comment (ionenwks) [inline app-vim/rust-vim/metadata.xml:16]: I don't see why this wouldn't be stabilize allarches, it only install .vim files unless something I don't know about.

File: app-vim/brainfuck-syntax/brainfuck-syntax-1.0.3-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (ionenwks) [inline app-vim/brainfuck-syntax/brainfuck-syntax-1.0.3-r2.ebuild:4]: I'd say these trivial EAPI bumps should be left to vim@ to handle (i.e. drop this commit). Back in April these received a mass bump from 5 -> 7 and I imagine 6 -> 7 (or even 8 if support is added) may happen eventually.

Also, for a trivial vim ebuild, I'd say revbump for a simple EAPI bump with no other changes accomplishes nothing but give more work to arch testers.

Comment (scardracs) [reply app-vim/brainfuck-syntax/brainfuck-syntax-1.0.3-r2.ebuild:4]: If you take a look on Manifest is not only EAPI update but also version update (from pre20180125 to pre20210215). If you prefer I can drop the -1.0.3-r2 and use release date

File: app-vim/rust-vim/metadata.xml [hunk 4]
```diff
@@ -1,7 +1,18 @@
...
 <pkgmetadata>
-	<!-- maintainer-needed -->
+	<maintainer type="person" proxied="yes">
+		<email>marco@scardovi.com</email>
+		<name>Marco Scardovi</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>vim@gentoo.org</email>
+		<name>Gentoo Vim Project</name>
+	</maintainer>
```
[the remark this answers is not in the document]
Comment (scardracs) [reply app-vim/rust-vim/metadata.xml:11]: Ok, I'll add it

File: app-vim/brainfuck-syntax/brainfuck-syntax-1.0.3-r1.ebuild [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
+EAPI=7
 
 inherit vim-plugin
 
-DESCRIPTION="vim plugin: BrainFuck source file syntax highlighting"
+DESCRIPTION="Support for the brainfuck programming language"
 HOMEPAGE="https://www.vim.org/scripts/script.php?script_id=716"
 LICENSE="BSD-2"
-KEYWORDS="amd64 ~mips ppc ppc64 x86"
+KEYWORDS="~amd64 ~mips ~ppc ~ppc64 ~x86"
```
Comment (ionenwks) [inline app-vim/brainfuck-syntax/brainfuck-syntax-1.0.3-r1.ebuild:11]: If no longer revbump'ing, keywords shouldn't be changed.

File: app-vim/rust-vim/rust-vim-1_pre20210215.ebuild [hunk 6]
```diff
@@ -4,16 +4,13 @@
...
-
-COMMIT="87c745d8d506fc1eecc1d81df15d5bde1658a2fc"
-MY_PN="${PN/-/.}"
+MY_COMMIT="87c745d8d506fc1eecc1d81df15d5bde1658a2fc"
 
 DESCRIPTION="Vim configuration for Rust"
 HOMEPAGE="https://www.rust-lang.org https://github.com/rust-lang/rust.vim"
-SRC_URI="https://github.com/rust-lang/rust.vim/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+SRC_URI="https://github.com/rust-lang/rust.vim/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${PN/-/.}-${COMMIT}"
```
Comment (ionenwks) [inline app-vim/rust-vim/rust-vim-1_pre20210215.ebuild:12]: As usual, don't go too quick :)

This is still using `COMMIT`

**Comment by ionenwks** [comment]
Hm, this new rust-vim version seems to be installing some unused test stuff.
```
/usr/share/vim/vimfiles/test/vimrc
/usr/share/vim/vimfiles/test/sample.rs
/usr/share/vim/vimfiles/test/run-tests
/usr/share/vim/vimfiles/test/indent.vader
/usr/share/vim/vimfiles/test/coverage.vader
/usr/share/vim/vimfiles/test/Dockerfile
/usr/share/vim/vimfiles/test/.gitignore
```

**Review by ionenwks — APPROVED** [review]
changing eapply_user -> default on merge as mentioned on IRC, but otherwise looks good

**Pull request closed without merging**